library/ssl_misc.h

```diff
@@ -1332,6 +1332,23 @@ void mbedtls_ssl_update_in_pointers( mbedtls_ssl_context *ssl );
 
 int mbedtls_ssl_session_reset_int( mbedtls_ssl_context *ssl, int partial );
 
+/*
+ * Send pending alert
+ */
+int mbedtls_ssl_handle_pending_alert( mbedtls_ssl_context *ssl );
+
+/*
+ * Set pending fatal alert flag.
+ */
+void mbedtls_ssl_pend_fatal_alert( mbedtls_ssl_context *ssl,
+                                   unsigned char alert_type,
+                                   int alert_reason );
+
+/* Alias of mbedtls_ssl_pend_fatal_alert */
+#define MBEDTLS_SSL_PEND_FATAL_ALERT( type, user_return_value )         \
+            mbedtls_ssl_pend_fatal_alert( ssl, type, user_return_value )
```

Comment:
Should it be modified to 
```c
#define MBEDTLS_SSL_PEND_FATAL_ALERT( type, user_return_value )    \
            do{                                                                                            \
                   mbedtls_ssl_pend_fatal_alert( ssl, type, user_return_value );  \
                   return( user_return_value ) ;                                                 \
           }while(0)
```
I think all fatal alert messages should be followed by `return` , right? And I prefer put `return` into the macro to avoid `alert_ret` is different with `ret` in `handshake_step`

Reply:
> I think all fatal alert messages should be followed by return , right? 

I'm not sure they'll all be followed _immediately_ by return: sometimes we might want to do some clean-up before returning. So, for now I'd keep the `return` out of the macro, and perhaps revisit later.

The idea with `##type` looks good, but  I don't think our naming of errors and alerts is consistent enough to support it, so again for now let's keep the two arguments and revisit later (perhaps even in 4.0 if public names have to be changed).

Reply:
Thanks @mpg . I just review all rest parts, my suggestion is wrong.